src/renderers/shared/reconciler/__tests__/ReactCompositeComponent-test.js

```diff
@@ -874,6 +874,124 @@ describe('ReactCompositeComponent', function() {
     expect(div.children[0].id).toBe('aliens');
   });
 
+  it('should trigger componentWillReceiveProps for context changes', function() {
+    var contextChanges = 0;
+    var propChanges = 0;
+
+    var GrandChild = React.createClass({
+      contextTypes: {
+        foo: ReactPropTypes.string.isRequired,
+      },
+
+      componentWillReceiveProps: function(nextProps, nextContext) {
+        expect('foo' in nextContext).toBe(true);
+
+        if (nextProps !== this.props) {
+          propChanges++;
+        }
+
+        if (nextContext !== this.context) {
+          contextChanges++;
+        }
+      },
+
+      render: function() {
+        return <span className="grand-child">{this.props.children}</span>;
+      },
+    });
+
+    var ChildWithContext = React.createClass({
+      contextTypes: {
+        foo: ReactPropTypes.string.isRequired,
+      },
+
+      componentWillReceiveProps: function(nextProps, nextContext) {
+        expect('foo' in nextContext).toBe(true);
+
+        if (nextProps !== this.props) {
+          propChanges++;
+        }
+
+        if (nextContext !== this.context) {
+          contextChanges++;
+        }
+      },
+
+      render: function() {
+        return <div className="child-with">{this.props.children}</div>;
+      },
+    });
+
+    var ChildWithoutContext = React.createClass({
+      componentWillReceiveProps: function(nextProps, nextContext) {
+        expect('foo' in nextContext).toBe(false);
+
+        if (nextProps !== this.props) {
+          propChanges++;
+        }
+
+        if (nextContext !== this.context) {
+          contextChanges++;
+        }
+      },
+
+      render: function() {
+        return <div className="child-without">{this.props.children}</div>;
+      },
+    });
+
+    var Parent = React.createClass({
+      childContextTypes: {
+        foo: ReactPropTypes.string,
+      },
+
+      getInitialState() {
+        return {
+          foo: 'abc',
+        };
+      },
+
+      getChildContext: function() {
+        return {
+          foo: this.state.foo,
+        };
+      },
+
+      onClick() {
+        this.setState({
+          foo: 'def',
+        });
+      },
+
+      render: function() {
+        return <div className="parent" onClick={this.onClick}>{this.props.children}</div>;
+      },
+    });
+
+    Parent.ChildWithContext = ChildWithContext;
+    Parent.ChildWithoutContext = ChildWithoutContext;
+    Parent.GrandChild = GrandChild;
+
+    var div = document.createElement('div');
+
+    ReactDOM.render(<Parent>
```

Comment:
nit: to match our usual style, can you add a linebreak after `(` and indent the next line two spaces? similarly, it should end

```
  </Parent>,
  div
);
```